jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryingLogGroupsTest.kt

```diff
@@ -0,0 +1,151 @@
+import java.util.Calendar
+
+@RunsInEdt
+class QueryingLogGroupsTest {
+    @JvmField
+    @Rule
+    val projectRule = JavaCodeInsightTestFixtureRule()
+    private val qEditorValidator = QueryEditorValidator
+    private lateinit var view: QueryEditor
+
+    @Before
+    fun setup() {
+        val project = projectRule.project
+        val sdk = IdeaTestUtil.getMockJdk18()
+        runInEdtAndWait {
+            runWriteAction {
+                ProjectJdkTable.getInstance().addJdk(sdk, projectRule.fixture.projectDisposable)
+                ProjectRootManager.getInstance(projectRule.project).projectSdk = sdk
+            }
+            view = QueryEditor(project)
+        }
+    }
+
+    @Test
+    fun absoluteOrRelativeTimeSelected() {
+        view.absoluteTimeRadioButton.isSelected = false
+        view.relativeTimeRadioButton.isSelected = false
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.validation.timerange"))
+    }
+
+    @Test
+    fun startDateNotSelected() {
+        view.absoluteTimeRadioButton.isSelected = true
+        view.startDate.date = null
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("Start Date must be specified")
+    }
+
+    @Test
+    fun endDateNotSelected() {
+        view.absoluteTimeRadioButton.isSelected = true
+        view.startDate.date = Calendar.getInstance().time
+        view.endDate.date = null
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("End Date must be specified")
+    }
+
+    @Test
+    fun startDateBeforeEndDate() {
+        view.absoluteTimeRadioButton.isSelected = true
+        view.startDate.date = Calendar.getInstance().time
+        val cal = Calendar.getInstance()
+        cal.add(Calendar.DATE, -1)
+        view.endDate.date = cal.time
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("Start date must be before End date")
+    }
+
+    @Test
+    fun relativeTimeUnitSelected() {
+        view.relativeTimeRadioButton.isSelected = true
+        view.relativeTimeUnit.selectedItem = "Minutes"
+        view.relativeTimeNumber.text = ""
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("Number must be specified")
+    }
+
+    @Test
+    fun searchOrQuerySelected() {
+        view.relativeTimeRadioButton.isSelected = true
+        view.startDate.date = null
+        view.endDate.date = null
+        view.absoluteTimeRadioButton.isSelected = false
+        view.relativeTimeUnit.selectedItem = "Minutes"
+        view.relativeTimeNumber.text = "1"
+        view.queryLogGroupsRadioButton.isSelected = false
+        view.searchTerm.isSelected = false
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("Query must be entered")
+    }
+
+    @Test
+    fun searchTermSpecified() {
+        view.relativeTimeRadioButton.isSelected = true
```

Comment:
All of these tests have very similar code. Something similar to `buildMockRdsNode(` where we build a view would have less duplication and be more clear

Reply:
I have added a function that sets the common parameters for each test.